### ISSUE TITLE
Deal with warnings generated by the `resolve_dates()` function

### ISSUE THREAD

**Issue by henriquesposito** [globalgov/qData#136]
The function works and returns observation in date formats, however, it generates several warnings when doing so.

**Comment by henriquesposito** [comment]
The function operates correctly but, sometimes, returns numeric date values rather than standard date formats. 

As, well, how the resolve family of functions are setup, there appear to be issues with if statements only taking first value.

**Comment by henriquesposito** [comment]
@BBieri could you please take a look at these warnings? I know you dealt with some of them beforehand when setting up the report functions. First, the functions works correctly, but when the date ranges are resolved a numeric date is returned instead of an a ymd date. Second, we get warnings when NAs are returned even though they are returned correctly and as expected. In any case, the tests for resolve functions are all setup and demonstrate these issues/warnings we get. Thank you!

**Comment by BBieri** [comment]
Hi @henriquesposito , I changed the dataframe of the first test to fit the actual data we'll have in the qPackages and corrected the remaining warnings that the resolve family of functions was generating. Let me know if you have any feedback on my changes.

An interesting fact I learned while doing the work was that it seems that the base `ifelse()` function does not retain classes hence I had to use the safer `if_else()` function from `{dplyr}`.
